.github/workflows/report.yml

```diff
@@ -0,0 +1,224 @@
+name: Docs, Coverage Report and PR updates
+
+# WARNING: This Workflow is executed with the workflow_run trigger, which means that 
+#          it is run in the context of the target/upstream repository and granted *write access*
+#          to the target/upstream repository.
+#          It *should only* be used for actions that require write access (e.g., to generate/deploy 
+#          the documentation/coverage report to GitHub Pages or to create a PR comment) and *must 
+#          never* execute code contained in the repository.
+
+on:
+  workflow_run:
+    workflows: ["Test and Analyze"]
+    types:
+      - completed
+
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_TEST_TIME_UNIT: 60,120
+  RUST_TEST_TIME_INTEGRATION: 60,120
+  RUST_TEST_TIME_DOCTEST: 60,120
+  BRANCH: |-
+    ${{
+      (github.event.workflow_run.head_repository.owner.login != github.event.workflow_run.repository.owner.login)
+        && format('{0}:{1}', github.event.workflow_run.head_repository.owner.login, github.event.workflow_run.head_branch)
+        || github.event.workflow_run.head_branch
+    }}
+  HEAD_REF: ${{ github.event.workflow_run.head_sha }}
+  REPO_URL: "${{ github.server_url }}/${{ github.repository }}"
+
+concurrency:
+  # env context is not available here, so i just copied the code for env.BRANCH.
+  group: ${{ github.workflow }}-${{
+      (github.event.workflow_run.head_repository.owner.login != github.event.workflow_run.repository.owner.login)
+        && format('{0}:{1}', github.event.workflow_run.head_repository.owner.login, github.event.workflow_run.head_branch)
+        || github.event.workflow_run.head_branch
+    }}
+  cancel-in-progress: true
+
+jobs:
+
+  docs:
+    runs-on: ubuntu-latest
+    outputs:
+      docs_dl: ${{ format('{0}/actions/runs/{1}/artifacts/{2}', env.REPO_URL, github.run_id, steps.upload-docs.outputs.artifact-id) }}
+      docs_url: |-
+        ${{
+          (github.event.workflow_run.head_repository.owner.login == github.event.workflow_run.repository.owner.login) 
+          && format('{0}/docs/{1}/libcoap_rs/', vars.DOCS_AND_COV_URL, env.BRANCH) 
+          || '' 
+        }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          ref: ${{ env.HEAD_REF }}
+      # --all-features uses GNUTLS als backend, must provide it.
+      - uses: awalsh128/cache-apt-pkgs-action@latest
+        with:
+          packages: libgnutls28-dev libgnutls30
+          version: 1.0
+      - uses: dtolnay/rust-toolchain@nightly
+        with:
+          components: rustc
+      - name: Build documentation
+        run: cargo doc --all-features --no-deps --workspace
+      - uses: actions/upload-artifact@v4
+        id: upload-docs
+        with:
+          name: docs
+          path: |
+            ./target/doc
+      # Deploy to GitHub Pages only if the PR is not from a forked repository.
+      # For security reasons, we don't want PRs from forks to upload coverage data to our GitHub Pages.
```

Comment:
Just out of interest, does this mean we run the CI pipeline for (external) PRs by default, without any prior approval?

Reply:
This is just out of additional caution, external PRs are also set to always require manual approval.